src/main/java/de/tum/cit/aet/artemis/core/service/course/CourseAccessService.java

```diff
@@ -0,0 +1,232 @@
+                .filter(course -> !user.getGroups().contains(course.getStudentGroupName())).collect(Collectors.toSet());
+    }
+
+    /**
+     * Unenroll a user from a course by removing them from the student group of the course
+     *
+     * @param user   The user that should get removed from the course
+     * @param course The course from which the user should be removed from
+     */
+    public void unenrollUserForCourseOrThrow(User user, Course course) {
+        authCheckService.checkUserAllowedToUnenrollFromCourseElseThrow(course);
+        userService.removeUserFromGroup(user, course.getStudentGroupName());
+        learnerProfileApi.ifPresent(api -> api.deleteCourseLearnerProfile(course, user));
+        final var auditEvent = new AuditEvent(user.getLogin(), Constants.UNENROLL_FROM_COURSE, "course=" + course.getTitle());
+        auditEventRepository.add(auditEvent);
+        log.info("User {} has successfully unenrolled from course {}", user.getLogin(), course.getTitle());
+    }
+
+    /**
+     * Enrolls a user in a course by adding them to the student group of the course
+     *
+     * @param user   The user that should get added to the course
+     * @param course The course to which the user should get added to
+     */
+    public void enrollUserForCourseOrThrow(User user, Course course) {
+        authCheckService.checkUserAllowedToEnrollInCourseElseThrow(user, course);
+        userService.addUserToGroup(user, course.getStudentGroupName());
+        if (course.getLearningPathsEnabled()) {
+            learnerProfileApi.ifPresent(api -> api.createCourseLearnerProfile(course, user));
+            learningPathApi.ifPresent(api -> api.generateLearningPathForUser(course, user));
+        }
+        final var auditEvent = new AuditEvent(user.getLogin(), Constants.ENROLL_IN_COURSE, "course=" + course.getTitle());
+        auditEventRepository.add(auditEvent);
+        log.info("User {} has successfully enrolled in course {}", user.getLogin(), course.getTitle());
+    }
+
+    /**
+     * Add multiple users to the course so that they can access it
+     * The passed list of UserDTOs must include at least one unique user identifier (i.e. registration number OR email OR login)
+     * <p>
+     * This method first tries to find the user in the internal Artemis user database (because the user is probably already using Artemis).
+     * In case the user cannot be found, it additionally searches the connected LDAP in case it is configured.
+     *
+     * @param courseId    the id of the course
+     * @param studentDTOs the list of students (with at least registration number)
+     * @param courseGroup the group the students should be added to
+     * @return the list of students who could not be enrolled in the course, because they could NOT be found in the Artemis database and could NOT be found in the TUM LDAP
+     */
```

Comment:
The LDAP part is true but something that happens two methods further down; I do not think that this implementation detail should be included in the javadoc of this method.

```suggestion
    /**
     * Add multiple users to the course so that they can access it.
     * The passed list of {@link StudentDTO} must include at least one unique user identifier (i.e. registration number OR email OR login)
     *
     * @param courseId    the id of the course
     * @param studentDTOs the list of students (with at least registration number)
     * @param courseGroup the group the students should be added to
     * @return the list of students who could not be enrolled in the course, because they could NOT be found in the Artemis database and could NOT be found in the LDAP
     * @see UserService#findUserAndAddToCourse
     * @see UserService#findUser
     */
```